js/app/demo/messages.jsx

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+define(['react', 'jsx!message'], function(React, Message) {
+    return React.createClass({
+        displayName: 'Messages',
+        render: function() {
+            if (!this.props.values || this.props.values.length === 0) {
+                return (
+                    <div id="results">
+                        <div className="alert alert-success">Lint-free!</div>
+                    </div>
+                )
+            } else {
+                return (
+                    <div id="results">
+                        {
+                            this.props.values.map(function(message, index) {
+                                return <Message key={index} value={message} />;
```

Comment:
Typically, I'd prefer to pass `line`, `column`, `ruleId`, and `message` as props directly on `Message`. This makes it slightly more convenient to access props within the component (don't need `.value`) and easier to validate via `PropTypes` (don't need to nest within `PropTypes.shape`). With a transpiler, I'd do that with `<Message key={key} {...message} />`. Without a transpiler, if `Object.assign` had slightly higher [compatibility](http://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign) or we had a polyfill, I'd do it with `React.createElement(Message, Object.assign({ key: key }, message))`. Since we have neither of those right now, I'm leaving a note and saying this is the best way to do it unless or until one of the above changes.